pkg/util/resize/configspec_pci_devices.go

```diff
@@ -26,34 +24,20 @@ func comparePCIDevices(
 	var deviceChanges []vimtypes.BaseVirtualDeviceConfigSpec
 	for _, expectedDev := range expectedPCIDevices {
 		expectedPci := expectedDev.(*vimtypes.VirtualPCIPassthrough)
-		expectedBacking := expectedPci.Backing
-		expectedBackingType := reflect.TypeOf(expectedBacking)
 
 		var matchingIdx = -1
 		for idx, curDev := range currentPassthruPCIDevices {
 			curBacking := curDev.GetVirtualDevice().Backing
-			if curBacking == nil || reflect.TypeOf(curBacking) != expectedBackingType {
+			if curBacking == nil {
 				continue
 			}
 
 			var backingMatch bool
-			switch a := curBacking.(type) {
+			switch a := expectedPci.Backing.(type) {
 			case *vimtypes.VirtualPCIPassthroughVmiopBackingInfo:
-				b := expectedBacking.(*vimtypes.VirtualPCIPassthroughVmiopBackingInfo)
-				backingMatch = a.Vgpu == b.Vgpu
-
+				backingMatch = MatchVirtualPCIPassthroughVmiopBackingInfo(a, curBacking)
 			case *vimtypes.VirtualPCIPassthroughDynamicBackingInfo:
-				currAllowedDevs := a.AllowedDevice
-				b := expectedBacking.(*vimtypes.VirtualPCIPassthroughDynamicBackingInfo)
-				if a.CustomLabel == b.CustomLabel && len(b.AllowedDevice) > 0 {
```

Comment:
So, if the labels matched, but the allowed devices didn't we skip the comparison?  Shouldn't the labels be the authority on identifying the correct device that is being updated?

Reply:
Talked offline - I cribbed this govmoni's SelectByBackingInfo() and makes sense that the label, if set, is sufficient in this case. I really only want to fix the blind `b.AllowedDevice[0]` assumption and can revisit later if needed.